<launch>
  <!-- machine file -->
  <include file="$(find pr2_machine)/$(env ROBOT).machine" />

  <!-- Buffer Server -->
  <node pkg="tf2_ros" type="buffer_server" name="cops_tf2_buffer_server" output="screen">
    <remap from="tf2_buffer_server" to="cops_tf2_buffer_server" />
    <param name="buffer_size" value="60.0"/>
  </node>

  <!-- ik action -->
  <node pkg="pr2_arm_move_ik" type="arm_ik" name="r_arm_ik" output="screen">
    <param name="joint_trajectory_action" value="r_arm_controller/joint_trajectory_generator" />
    <param name="arm" value="r" />
    <param name="free_angle" value="2" />
    <param name="search_discretization" value="0.01" />
    <param name="ik_timeout" value="5.0" />
  </node>

  <!-- Trajectory generator -->
  <node pkg="joint_trajectory_generator" type="joint_trajectory_generator" output="screen"
        name="r_joint_trajectory_generator" ns="r_arm_controller" >
    <param name="max_acc" value="2.0" />
    <param name="max_vel" value="2.5" />
  </node>
  <node pkg="joint_trajectory_generator" type="joint_trajectory_generator" output="screen"
        name="l_joint_trajectory_generator" ns="l_arm_controller" >
    <param name="max_acc" value="2.0" />
    <param name="max_vel" value="2.5" />
  </node>

  <!-- navstack -->
  <include file="$(find wg_pr2_2dnav)/pr2_2dnav.launch" />

  <param name="move_base_node/oscillation_timeout" value="300.0" />
  <param name="move_base_node/oscillation_distance" value="0.5" />

  <param name="move_base_node/base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
  <param name="move_base_node/sbpl_recovery/sbpl_lattice_planner/primitive_filename" value="$(find sbpl)/matlab/mprim/pr2.mprim" />
  <rosparam file="$(find pr2_base_application)/config/recovery_behaviors.yaml" command="load" ns="move_base_node" />
  <rosparam file="$(find pr2_base_application)/config/costmap_config.yaml" command="load" ns="move_base_node" />
  <rosparam file="$(find pr2_base_application)/config/dwa_local_planner.yaml" command="load" ns="move_base_node/DWAPlannerROS" />

  <!-- recording -->
  <node name="record_general_continuous_info" pkg="rosbag" type="record" args="-O /removable/continuous_operation/continuous_log /base_odometry/odometer /tf /join\
t_states /base_controller/command /power_state" />

</launch>
